vowpalwabbit/mf.cc

```diff
@@ -59,8 +57,12 @@ void predict(mf& data, single_learner& base, example& ec)
   ec.indices.clear();
   ec.indices.push_back(0);
 
+  auto* saved_interactions = ec.interactions;
```

Comment:
use a `swap_guard` since this fn is a bit more complex than the other ones where we do this